sourcegraph/cody
# VSCode: Add a testing flag to override local embeddings model dimension

This PR adds support for Sourcegraph-managed Embedding generation API - see https://github.com/sourcegraph/cody/pull/3753 for more context behind the changes.

## Test plan

- See test plan from https://github.com/sourcegraph/cody/pull/3753 (which was merged into this)
- Build cody-engine with 9285e0b, for example `cd crates/bfg && cargo build --release`
- Set the following user settings:

```
  "cody.experimental.cody-engine.path": "/path/to/release/bfg",
  "cody.testing.localEmbeddings.model": "sourcegraph/triton",
  "cody.testing.localEmbeddings.dimension": 768,
  "cody.testing.localEmbeddings.indexLibraryPath": "/tmp/indexes",
```

- Launch the extension with `CODY_TESTING=true` in the environment. For example, edit .vscode/launch.json's desktop configuration env hash to include `"CODY_TESTING": "true"`. Note, in this mode some storage is not persistent and you will have to log in each time.
- Open a git repo, start a chat, hit Enable Embeddings.
- After indexing finishes (or embed some stuff, then restart to use a partial index), chat and verify that Output > Cody by Sourcegraph contains embedding results, etc.
